From f0cf4495fc1d4dfb71a93eb6e315789c276221da Mon Sep 17 00:00:00 2001
From: Bin Liu <b-liu@ti.com>
Date: Wed, 30 May 2012 15:35:07 -0500
Subject: [PATCH 1/2] usb: musb: am35x: set default VBUS timeout value

The issue was:
musb->a_wait_bcon was not set so the OTG timer does not start when
a USB device gets disconnected to reset the OTG back to b_idle state.

Signed-off-by: Bin Liu <b-liu@ti.com>
---
 drivers/usb/musb/am35x.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/drivers/usb/musb/am35x.c b/drivers/usb/musb/am35x.c
index 57d4417..a59a5e7 100644
--- a/drivers/usb/musb/am35x.c
+++ b/drivers/usb/musb/am35x.c
@@ -79,6 +79,8 @@
 #define AM35X_TX_INTR_MASK	(AM35X_TX_EP_MASK << AM35X_INTR_TX_SHIFT)
 #define AM35X_RX_INTR_MASK	(AM35X_RX_EP_MASK << AM35X_INTR_RX_SHIFT)
 
+#define A_WAIT_BCON_TIMEOUT     1100            /* in ms */
+
 /* CPPI 4.1 queue manager registers */
 #define QMGR_PEND0_REG		0x4090
 #define QMGR_PEND1_REG		0x4094
@@ -599,6 +601,7 @@ static int am35x_musb_init(struct musb *musb)
 	cppi41_init(musb);
 #endif
 
+	musb->a_wait_bcon = A_WAIT_BCON_TIMEOUT;
 	musb->isr = am35x_musb_interrupt;
 
 	/* clear level interrupt */
-- 
1.7.0.4

